sdk/core/core-sse/test/public/node/connection.spec.ts

```diff
@@ -88,7 +87,7 @@ describe("[Node] Connections", () => {
               ran = true;
               if (sse.data === "[DONE]") {
                 if (path.includes("no-fin")) {
-                  assert.isNull(stream.socket);
+                  assert.isNull((stream as any).socket);
```

Comment:
if socket is required, should we define the input shape to include this as well? Or what would happen if someone passed us a `fs.createReadableStream` instead?

Reply:
`socket` is part of `IncomingMessage` which is already an input type to the `createSseStream` function. If someone, passes a non-IncomingMessage object that doesn't have a socket, `destroy` will be called instead. The reason I am going with `socket.end()` first is that it is a more graceful way to close the connection because it sends a `FIN` packet to the server and flushes what is currently in the pipeline. On the other hand,  `destroy` is more forceful as it shuts down the socket immediately.